### DISCUSSION TITLE
TCP Checksum Issue

### DISCUSSION THREAD

**Discussion by JaesukLim** [ANLAB-KAIST/KENSv3#208] [Project Q&A 2. Transport Layer (TCP)]
I started to implement write() with TestTransfer_Connect_Send_Symmetric
However, when I send packet with write(), it shows checksum error
![image](https://github.com/ANLAB-KAIST/KENSv3/assets/143942994/bbd679fd-c435-4544-8c42-b5fb70a85c02)
Actually, the checksum in ACK packet in handshake was correct. 
The only difference in calculating checksum of the packet 3 (handshake ACK) and packet 4 (by write()) is the length, right? (20 vs 20 + 512)
Is there anything I missing during calculating checksum?

By the way, is there recommended way to implement or order of test cases?
Thank you

**Comment by Pusnow** [comment]
It seems that your packet is truncated. Did you resize the Packet structure correctly?

**Reply by JaesukLim** [reply]
Oh in this case the write call wants to send 1024 bytes but at the document there was 'set MTU size as 552' so I truncated with 512. Did I misinterpreted?

**Reply by Pusnow** [reply]
I mean the size of `Packet` data structures.

You can change or get size of the data structures using:

https://github.com/ANLAB-KAIST/KENSv3/blob/5bf97d4d228d6a6bcb8a2fdc1f50abe9c075c701/include/E/Networking/E_Packet.hpp#L93-L104

**Reply by JaesukLim** [reply]
Oh I just didn't know about when calculating checksum, tcp_seg field should contain the actual data. Thank you
